notifier/src/main/kotlin/Notifier.kt

```diff
@@ -43,15 +43,20 @@ class Notifier(
 ) : ScriptRunner() {
     private val customProperties = buildMap {
         put("ortResult", ortResult)
-
-        config.mail?.let { put("mailClient", MailNotifier(it)) }
-        config.jira?.let { put("jiraClient", JiraNotifier(it)) }
-
+        put("mailClient", config.mail?.let { MailNotifier(it) })
+        put("jiraClient", config.jira?.let { JiraNotifier(it) })
         put("resolutionProvider", resolutionProvider)
     }
 
+    private val customPropertyTypes = mapOf(
+        "ortResult" to KotlinType(OrtResult::class),
+        "mailClient" to KotlinType(MailNotifier::class, isNullable = true),
+        "jiraClient" to KotlinType(JiraNotifier::class, isNullable = true),
```

Comment:
Making these types nullable will be a breaking change because all notifier scripts will need to be adjusted.
Please change your commit message: `fix(notifier)!:`